pilot/pkg/model/jwks_resolver.go

```diff
@@ -435,3 +435,41 @@ func (r *JwksResolver) refresh() {
 func (r *JwksResolver) Close() {
 	closeChan <- true
 }
+
+func compareJWKSResponse(oldKeyString string, newKeyString string) bool {
+	var oldJWKs map[string]interface{}
+	var newJWKs map[string]interface{}
+	e := json.Unmarshal([]byte(oldKeyString), &oldJWKs)
+	if e != nil {
+		log.Warnf("Previous JWKs public key JSON is not parseable: %s", oldKeyString)
+		return true
+	}
+	e = json.Unmarshal([]byte(newKeyString), &newJWKs)
+	if e != nil {
+		log.Warnf("New JWKs public key JSON is not parseable: %s", newKeyString)
+		return true
+	}
+
+	if oldJWKs == nil || newJWKs == nil {
```

Comment:
Just curious, when these can be nil? Also, if both are nil, should it return false (they are 'equal')?

Reply:
With the unmarshalling and catching the error there I don't think these can be nil, I'm going to remove this check.